lib/blacklight/canvas_course.rb

```diff
@@ -54,7 +65,72 @@ def self.from_metadata(metadata)
             name: course_name,
           },
         )
-      CanvasCourse.new(metadata, canvas_course)
+      CanvasCourse.new(metadata, canvas_course, blackboard_export)
+    end
+
+    ##
+    # Creates a canvas assignment from a scorm package that has already been
+    # uploaded to a scorm manager
+    ##
+    def create_scorm_assignment(scorm_package, course_id)
+      url = "#{Blacklight.scorm_launch_url}?" +
+        "course_id=#{scorm_package['package_id']}"
+
+      payload = {
+        assignment__submission_types__: ["external_tool"],
+        assignment__integration_id__: scorm_package["package_id"],
+        assignment__integration_data__: {
+          provider: "atomic-scorm",
+        },
+        assignment__external_tool_tag_attributes__: {
+          url: url,
+        },
+      }
+
+      CanvasCourse.client.create_assignment(
+        course_id,
+        scorm_package["title"],
+        payload,
+      )
+    end
+
+    ##
+    # Uploads a scorm package to scorm manager specified in blacklight.yml
+    # config file
+    ##
+    def upload_scorm_package(scorm_package, course_id, tmp_name)
+      zip = scorm_package.write_zip tmp_name
+      RestClient.post(
+        "#{Blacklight.scorm_url}/api/scorm_courses",
+        {
+          oauth_consumer_key: "scorm-player",
+          lms_course_id: course_id,
+          file: File.new(zip, "rb"),
```

Comment:
Referencing `File.new` this way opens a file, but never closes it. Better practice would be to call `File.open` with a block, and put the `RestClient.post` inside the block, like this:

```ruby
File.open(zip, "rb") do |file|
  RestClient.post(
      ...,
      file: file,
      ...,
  )
  # etc.
end
```

This will ensure that the file is automatically closed when the block terminates. (Ruby takes care of a lot of things for you, but managing IO still requires some attention, alas!)